smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/xml/MemberShapeDecodeXMLGenerator.kt

```diff
@@ -293,14 +293,14 @@ abstract class MemberShapeDecodeXMLGenerator(
         renderAssigningDecodedMember(memberName, "$value")
     }
 
-    fun renderScalarMember(member: MemberShape, memberTarget: Shape, containerName: String, unkeyed: Boolean = false) {
+    fun renderScalarMember(member: MemberShape, memberTarget: Shape, containerName: String, unkeyed: Boolean = false, isUnion: Boolean = false) {
```

Comment:
1.  I'm surprised that protocol codegen tests still pass with or without this change. I think we should add at least one protocol codegen test.  Clearly, we'll need one that addresses the decode, but I suspect we'll want to add one for encode as well.

Reply:
can you elaborate further on what that protocol codegen test might be testing for? in this case, it seems we got the spec wrong and when its a union, its value can never be null. so we should never be using decodeIfPresent if we are decoding the associated value of an enum?

Reply:
I think a unit test might suffice in this case

Reply:
I think i misunderstood the changes, yep, lgtm!